Assets/Countly/Plugins/CountlySDK/Services/SessionCountlyService.cs

```diff
@@ -71,26 +73,38 @@ internal async Task StartSessionService()
         /// </summary>
         private void InitSessionTimer()
         {
-            _sessionTimer = new Timer { Interval = _configuration.SessionDuration * 1000 };
+            _sessionTimer = new Timer { Interval = _configuration.GetUpdateSessionTimerDelay() * 1000 };
             _sessionTimer.Elapsed += SessionTimerOnElapsedAsync;
             _sessionTimer.AutoReset = true;
             _sessionTimer.Start();
         }
 
+        /// <summary>
+        /// Stops extending a session
+        /// </summary>
+        internal void StopSessionExtending()
+        {
+            _isSessionExtendingStopped = true;
```

Comment:
remove everything else and just call _sessionTimer.Stop() here
also name it stopIternalTimer or stopInternalTick or sth, because even if there is no session it would work

Reply:
sadly this doesn't solve our problem just by itself. to get as close as possible to your request, just added a cancellation token, renamed the method and called cancel too in the method. early exit needs to be in the SessionTimerOnElapsedAsync if we are not going to rework the whole thing

Reply:
It does not seem like reworking the whole thing is necessary. Stopping the timer does not stop the SessionTimerOnElapsedAsync call that was happening at that time and that seem to be the issue you are mentioning. Then you introduced the cancel token which brings in more questions. This comment mentions it is dangerous to not throw a cancellation requested: https://stackoverflow.com/questions/7343211/cancelling-a-task-is-throwing-an-exception/7343338#comment53310425_7343338 also a link from there: https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/how-to-cancel-a-task-and-its-children?redirectedfrom=MSDN
What are your thoughts about this? Read them and let me know.